lib/helpers/utils.js

```diff
@@ -11803,25 +11810,44 @@ export function getGradleCommand(srcPath, rootPath) {
  * @param {string[]} gradleArguments The general gradle arguments, which must only be added once
  * @param {string[]} gradleSubCommands The sub-commands that are to be executed by gradle
  * @param {string[]} gradleSubCommandArguments The arguments specific to the sub-command(s), which much be added PER sub-command
+ * @param {int} gradleCommandLength The length of the full gradle-command
  *
- * @returns {string[]} Array of arguments to be added to the gradle command
+ * @returns {string[]} Array of arrays of arguments to be added to the gradle command
  */
 export function buildGradleCommandArguments(
   gradleArguments,
   gradleSubCommands,
   gradleSubCommandArguments,
+  gradleCommandLength,
 ) {
-  let allGradleArguments = [
+  const mainGradleArguments = [
     "--build-cache",
     "--console",
     "plain",
     "--no-parallel",
   ].concat(gradleArguments);
+  const maxCliArgsLength = isWin
+    ? 7500 - gradleCommandLength - mainGradleArguments.join(" ").length - 2
```

Comment:
Is 7500 the maximum gradle supports on windows?

Reply:
I am fine with creating a temporary powershell script and executing the same if it could yield good performance instead of batching.

Reply:
I hadn't thought about that yet. Let me check that out, it might make this, and your question above, moot.

Reply:
I just can't figure out how to get powershell to return the output... @prabhu, do you have any idea/experience with powershell?

Reply:
@malice00, powershell supports redirecting [output](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/out-file?view=powershell-7.5) to files.